c7n/actions.py

```diff
@@ -186,26 +186,115 @@ class ModifyVpcSecurityGroupsAction(Action):
         'properties': {
             'type': {'enum': ['modify-security-groups']},
             'add': {'oneOf': [
-                {'type': 'string', 'pattern': '^sg-*'},
+                {'type': 'string'},
                 {'type': 'array', 'items': {
-                    'pattern': '^sg-*',
                     'type': 'string'}}]},
             'remove': {'oneOf': [
                 {'type': 'array', 'items': {
-                    'type': 'string', 'pattern': '^sg-*'}},
+                    'type': 'string'}},
                 {'enum': [
                     'matched', 'network-location', 'all',
-                    {'type': 'string', 'pattern': '^sg-*'}]}]},
+                    {'type': 'string'}]}]},
             'isolation-group': {'oneOf': [
-                {'type': 'string', 'pattern': '^sg-*'},
+                {'type': 'string'},
                 {'type': 'array', 'items': {
-                    'type': 'string', 'pattern': '^sg-*'}}]}},
+                    'type': 'string'}}]}},
         'anyOf': [
             {'required': ['isolation-group', 'remove', 'type']},
             {'required': ['add', 'remove', 'type']},
             {'required': ['add', 'type']}]
     }
 
+    def resolve_security_group_names(self, group_names):
+        """
+        Resolves Security Group names to Ids.
+        Raises PolicyExecutionError if group names are not found
+        """
+        if group_names:
+            client = utils.local_session(
+                self.manager.session_factory).client('ec2')
+
+            filtered_sgs = client.describe_security_groups(
+                Filters=[
+                    {
+                        'Name': 'group-name',
+                        'Values': group_names
+                    }
+                ]
+            )['SecurityGroups']
+
+            filtered_ids = [a['GroupId'] for a in filtered_sgs]
+
+            if not filtered_ids or len(filtered_ids) != len(group_names):
```

Comment:
If you have multiple VPCs in a region and you have a security group with the same group name in each VPC, wouldn't this check always fail? `len(filtered_ids) != len(group_names)`